### PR TITLE
[RNMobile] - test break tags removal

### PR DESCRIPTION
**Pull request by cameronvoell**
## Description

Fixes part of https://github.com/wordpress-mobile/gutenberg-mobile/issues/1543 addressing some comments from here: https://github.com/WordPress/gutenberg/pull/18138#pullrequestreview-308836159

This changes adds more visibility into selection adjustments that we are making in gutenberg mobile code to ensure compatibility with react-native-aztec.

The code tested in this change is related to addressing a crash that was surfaced by the WordPressAndroid app which is using this code: https://github.com/wordpress-mobile/WordPress-Android/issues/9832

## How has this been tested?
Tested using steps from the [PR](https://github.com/wordpress-mobile/gutenberg-mobile/pull/1507) that originally added this code:

1. Create a post on Gutenberg web with a para block that ends with one or more empty line(s) (Shift+Enter)
2. Open this new post in GB mobile
3. Tap on the Plus symbol to add a new block below the para
4. Put the focus into this new block and taps on Backspace to merge this new block with the one above.
5. It should not crash, and the 2 blocks merged, with the caret at the end of the block

Also you can run the new tests using the following:
`yarn test gutenberg/packages/rich-text/src/component/test/index.native.js`

## Types of changes

This change is only introducing some new unit tests and a small refactor to accommodate those tests. 

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR. .

### CODE REVIEW

File: packages/rich-text/src/component/index.native.js [hunk 1]
```diff
@@ -649,6 +650,16 @@ export class RichText extends Component {
 		return value;
 	}
 
+	countBreaksBeforeParagraphTag( html ) {
```
Comment (mchowning) [inline packages/rich-text/src/component/index.native.js]: What do you think about specifying that it's an end tag: `countBreaksBeforeParagraphEndTag`? I don't feel strongly about this.

File: packages/rich-text/src/component/index.native.js [hunk 2]
```diff
@@ -693,23 +704,18 @@ export class RichText extends Component {
...
-							selection = { start: newSelectionStart, end: newSelectionEnd };
+						let newSelectionStart = this.props.selectionStart - count;
+						if ( newSelectionStart < 0 ) {
+							newSelectionStart = 0;
+						}
+						let newSelectionEnd = this.props.selectionEnd - count;
+						if ( newSelectionEnd < 0 ) {
+							newSelectionEnd = 0;
 						}
+						selection = { start: newSelectionStart, end: newSelectionEnd };
```
Comment (mchowning) [inline packages/rich-text/src/component/index.native.js]: I really like how you have the regex broken out into a separate function with tests (those tests you added are 😻). 

I would have probably taken this a bit further and extracted out all or most of 707-718 into some kind of pure 'calculateSelectionBasedOnAztecModifications( html, { originalStartSelection, originalEndSelection } )` function that took in the html and old selection and output the updated selection for the anticipated Aztec-output text. We would then end up with just having something like this in the `render` method:

```
...
} else {
    selection = calculateSelectionBasedOnAztecModifications( html, selection );
}
```

With that said, I think extracting that out at this point falls well within the realm of personal preference, and the changes you have made are already a really great improvement, so feel free to leave this as-is if you prefer. 😄

**Review by mchowning — APPROVED** [review]
Nice job @cameronvoell ! Those tests make the intention here sooo much clearer imo! 🙇 

I left a couple of minor comments, but they're really just suggestions/thoughts. This looks good to me! 🎉

**Comment by hypest** [comment]
This PR looks like it's stale. Can you circle back @cameronvoell and either land it or close it? Thanks!

**Comment by gziolo** [comment]
This PR looks to be abandoned, I marked it as stale. If it won’t get updated in the next few weeks it might get closed.

It was approved with minor comments so it seems beneficial to rebase it and take to the finish line.

**Comment by Thelmachido** [comment]
@cameronvoell  This issue was marked as stale in 2020 it has seen no movement since then, closing this PR for now. If anyone feels it should still be looked at, please add a comment and I can take another look. Thanks!

**Pull request closed without merging**